src/models/db.js

```diff
@@ -304,6 +330,13 @@ module.exports = config => {
         // For each possible association we check if the request body contains
         // any reference to the association model.
         if (!body) {
```

Comment:
I guess here is where the issue about creating the FeatureOfInterest from the associated Location if it is missing should be implemented. So add the comment here as well, please.